core/trino-main/src/main/java/io/trino/security/InjectedConnectorAccessControl.java

```diff
@@ -453,6 +459,29 @@ public void checkCanExecuteFunction(ConnectorSecurityContext context, SchemaRout
         accessControl.checkCanExecuteFunction(securityContext, new QualifiedObjectName(catalogName, function.getSchemaName(), function.getRoutineName()));
     }
 
+    @Override
+    public void checkCanExecuteFunction(ConnectorSecurityContext context, SchemaRoutineName function, FunctionType functionType)
+    {
+        checkArgument(context == null, "context must be null");
+        accessControl.checkCanExecuteFunction(securityContext, new QualifiedObjectName(catalogName, function.getSchemaName(), function.getRoutineName()), toFunctionKind(functionType));
+    }
+
+    private static FunctionKind toFunctionKind(FunctionType functionType)
```

Comment:
Maybe define a method in `FunctionType`?

Reply:
there are no more conversions